dll/shellext/zipfldr/CZipExtract.cpp

```diff
@@ -344,6 +353,188 @@ class CZipExtract :
         PropertySheetW(&psh);
     }
 
+    eZipExtractError ExtractSingle(
+        HWND hDlg,
+        LPCSTR FullPath,
+        bool is_dir,
+        unz_file_info64* Info,
+        CStringA Name,
+        CStringA Password,
+        bool* bOverwriteAll,
+        const bool* bCancel
+    )
+    {
+        int err;
+        BYTE Buffer[2048];
+        DWORD dwFlags = SHPPFW_DIRCREATE | (is_dir ? SHPPFW_NONE : SHPPFW_IGNOREFILENAME);
+        HRESULT hr = SHPathPrepareForWriteA(hDlg, NULL, FullPath, dwFlags);
+        if (FAILED_UNEXPECTEDLY(hr))
+        {
+            SetLastError(hr);
```

Comment:
Ok, switched to variable instead of Get/SetLastError. Regarding `HRESULT` I have additional checks in message prompt function, that's why I return it as is without conversion.